File: IGEC/admin/src/main/java/com/igec/admin/Activities/LauncherActivity.java
package com.igec.admin.Activities;


import static com.igec.common.CONSTANTS.EMPLOYEE_COL;
import static com.igec.common.CONSTANTS.ID;
import static com.igec.common.CONSTANTS.IGEC;

import android.app.Activity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;

import androidx.annotation.Nullable;

import com.google.firebase.firestore.FirebaseFirestore;

public class LauncherActivity extends Activity {
    private final FirebaseFirestore db = FirebaseFirestore.getInstance();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                SharedPreferences preferences = getSharedPreferences(IGEC, MODE_PRIVATE);
                if (preferences.getString(ID, "").equals("")) {
                    EMPLOYEE_COL.whereEqualTo("admin", true).limit(1).get().addOnSuccessListener(docs -> {
                        if (docs.size() != 0) {
                            Intent intent = new Intent(LauncherActivity.this, com.igec.admin.Activities.LoginActivity.class);
                            startActivity(intent);
                        } else {
                            Intent intent = new Intent(LauncherActivity.this, MainActivity.class);
                            startActivity(intent);
                        }
                        finish();
                    });
                } else {
                    Intent intent = new Intent(LauncherActivity.this, MainActivity.class);
                    startActivity(intent);
                    finish();
                }
            }

        }, 2000);


    }
}

File: IGEC/admin/src/main/java/com/igec/admin/activities/LauncherActivity.kt
package com.igec.admin.activities

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.Network
import android.os.Bundle
import android.os.Handler
import com.igec.common.CONSTANTS
import com.google.firebase.firestore.QuerySnapshot

class LauncherActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val connectivityManager = applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        connectivityManager.let {
            it.registerDefaultNetworkCallback(object : ConnectivityManager.NetworkCallback() {
                override fun onAvailable(network: Network) {
                    //take action when network connection is gained
                    Handler().postDelayed({
                        val preferences = getSharedPreferences(CONSTANTS.IGEC, MODE_PRIVATE)
                        if (preferences.getString(CONSTANTS.ID, "") == "") {
                            CONSTANTS.EMPLOYEE_COL.whereEqualTo("admin", true).limit(1).get()
                                .addOnSuccessListener { docs: QuerySnapshot ->
                                    if (docs.size() != 0) {
                                        val intent = Intent(this@LauncherActivity, LoginActivity::class.java)
                                        startActivity(intent)
                                    } else {
                                        val intent = Intent(this@LauncherActivity, MainActivity::class.java)
                                        startActivity(intent)
                                    }
                                    finish()
                                }
                        } else {
                            val intent = Intent(this@LauncherActivity, MainActivity::class.java)
                            startActivity(intent)
                            finish()
                        }
                    }, 0)
                }

                override fun onLost(network: Network) {
                    //take action when network connection is lost
                    val intent = Intent(this@LauncherActivity, InternetConnection::class.java)
                    startActivity(intent)
                    finish()
                }
            })
        }
    }
}